commerce-payment-method-paypal/src/main/java/com/liferay/commerce/payment/method/paypal/internal/PayPalCommercePaymentMethod.java

```diff
@@ -847,16 +837,73 @@ private ItemList _getItemList(
 
 			item.setName(commerceOrderItem.getName(languageId));
 			item.setPrice(
-				_payPalDecimalFormat.format(commerceOrderItem.getUnitPrice()));
+				_payPalDecimalFormat.format(commerceOrderItem.getFinalPrice()));
 			item.setQuantity(String.valueOf(commerceOrderItem.getQuantity()));
 			item.setSku(commerceOrderItem.getSku());
 
 			items.add(item);
 		}
 
+		_getOrderDiscounts(commerceOrder, commerceCurrency, locale, items);
+
+		_getShipping(commerceOrder, commerceCurrency, locale, items);
+
 		return items;
 	}
 
+	private String _getLabel(Locale locale, String key) {
+		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+			"content.Language", locale, getClass());
+
+		return LanguageUtil.get(resourceBundle, key);
+	}
+
+	private void _getOrderDiscounts(
+		CommerceOrder commerceOrder, CommerceCurrency commerceCurrency,
+		Locale locale, List<Item> items) {
```

Comment:
don't use references object as paraemter. Instead return a new list of discounted items